### PR TITLE
Basic contribution guide

### PR DESCRIPTION
**Pull request by SakiiR**
This is open to review/change

### CODE REVIEW

**Comment by SakiiR** [comment]
We can also add this as a wiki page. For now, and for review purpose, it's a new file

**Review by JJ-8 — CHANGES_REQUESTED** [review]
You mention that you have to fork it after the installation steps. I think this should be before the installation so you don't have to mess with the git settings to change the repo.

**Comment by SakiiR** [comment]
I moved the "git process" part on top of the document

**Review by JJ-8 — APPROVED** [review]
Yes, this is much better! :)

It is a quite basic guide, but I don't know what to add more. It seems you have covered all the steps already!

@B-i-t-K or @XeR, do you have any additions?

**Comment by SakiiR** [comment]
Thanks !

I think in the future, we could add some instruction on:

- Add a new migration
- Flushing the database
- Add a new page/component to the front app
- Implement a new query/migration

And other specific things about the technologies we use.

**Comment by JJ-8** [comment]
Ah, yes, these are very good additions. But do we want to include them now? Or should this PR just be the basics and we can extend it later?

**Comment by SakiiR** [comment]
> Ah, yes, these are very good additions. But do we want to include them now? Or should this PR just be the basics and we can extend it later?

Maybe later yes =)

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# CTFNote - Contribution Guide
+
+When contributing to this repository, please first discuss the change you wish to make via issue with the owners/moderators of this repository before making a change.
```
Comment (XeR) [inline CONTRIBUTING.md:3]: What's a moderator?
I think we should use the more generic term "contributors" or "collaborators". The later is what Github uses.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+- `132-add-new-feature`
+- `343-add-past-ctf-role`
+- `13-fix-bug-in-password-reset`
+- `37-leak-flag-to-TFNS`
+- `0-contribution-guide`
+
+Once you think the job is done, issue the pull request and target the **dev** branch of the official CTFNote repository.
+
+You can also create the pull request before finishing the job but don't forget to add "WiP: " as a suffix in the title to let the moderators know you are still working on the request.
```
Comment (XeR) [inline CONTRIBUTING.md:27]: cf. previous comment regarding "moderators"

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+The following endpoint are exposed and can be used in the developpement environment
+
+- [API](http://localhost:3000/)
+- [GraphiQL](http://localhost:3000/graphiql)
+- [Hedgedoc](http://localhost:3001/)
+- [Quasar APP](http://localhost:8088/)
+- [Adminer](http://localhost:3002/?pgsql=db&username=ctfnote&db=ctfnote)
+
+### Review
```
Comment (XeR) [inline CONTRIBUTING.md:90]: Should be a level 2 header

**Review by XeR — CHANGES_REQUESTED** [review]
Minor details about "moderators" and header depth.


Regarding the 2-man review requirements : does the PR submitter count as one if they are an established contributor ?
e.g. if BitK or JJ-8 makes a PR, does it start at 1/2 ?
I'm thinking about the few times where BitK and I did pair programming : he would push stuff, I would test it and approve it to get bugs fixed faster.

**Comment by JJ-8** [comment]
> Regarding the 2-man review requirements : does the PR submitter count as one if they are an established contributor ?

Actually, it is a 1-man review requirement, right? For example, I have just approved #131 and now I can click to merge it. So only one reviewer is required (where the author does not count as a reviewer). However, I never click 'merge PR' since I think it is better if someone else also looks at the PR (like this PR).

**Comment by JJ-8** [comment]
Maybe something else that is good to add at some point: where should a PR point to by default? The dev branch or the main branch?

**Comment by JJ-8** [comment]
Oh, and one more thing: maybe mentioning somewhere that migrations can not be changed after commitment (like [this](https://github.com/TFNS/CTFNote/pull/124#pullrequestreview-827792374) situation).

**Comment by XeR** [comment]
> Actually, it is a 1-man review requirement, right?

Yes. It's currently configured at 1 reviewer for main and dev. Administrators can force merge on dev only.


> where should a PR point to by default?

This is addressed in the "Git process" part :

>  Once you think the job is done, issue the pull request and target the **dev** branch of the official CTFNote repository.


> maybe mentioning somewhere that migrations can not be changed after commitment 

Good idea !

**Review by XeR — APPROVED** [review]

**Comment by XeR** [comment]
Merged manually in 0a17f424e282f48de0a18a80a7affbb4231dab14

**Pull request closed without merging**